src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java

```diff
@@ -76,13 +80,49 @@ public JSONArray filterFormParam(JSONObject criteriaObj, JSONObject optionsObj)
             String result = "{ \"$and\" : [ " + criteria + "," + options + " ] }";
             resultAggregatedObject = processAggregatedObject.processQueryAggregatedObject(result, databaseName, aggregationCollectionName);
         }
-        LOGGER.debug("resultAggregatedObject : " + resultAggregatedObject.toString());
+        if (filterKey == null || filterKey.toString().equals("{}")) {
+            LOGGER.debug("resultAggregatedObject : " + resultAggregatedObject.toString());
+        } else {
+            resultAggregatedObject = filterResult(resultAggregatedObject, filterKey);
+            LOGGER.debug("Filtered values from resultAggregatedObject : " + resultAggregatedObject.toString());
+        }
         return resultAggregatedObject;
     }
 
     /**
-     * This method takes the parameters from the REST GET request query. If the
-     * Aggregated Object matches the condition, then it is returned.
+     * This method takes array of aggregated objects and a filterKey. It returns a JSONArray where each element has a key (object Id)
+     * and a list of filtered values.
+     *
+     * @param resultAggregatedObjectArray,filterKey
+     * @return JSONArray
+     * @throws IOException
+     */
+    private JSONArray filterResult(JSONArray resultAggregatedObjectArray, JSONObject filterKey) {
+        JSONArray resultArray = new JSONArray();
+        JmesPathInterface unitUnderTest = new JmesPathInterface();
```

Comment:
Is this a unit under test in the running code? :)

Reply:
Wrong variable name. Fixed